core/store/src/db.rs

```diff
@@ -380,6 +509,7 @@ impl Database for RocksDB {
     }
 
     fn write(&self, transaction: DBTransaction) -> Result<(), DBError> {
+        self.pre_write_check().unwrap();
```

Comment:
Since we can catch disk low, we can doing a graceful shutdown of node here, instead of panic?

Reply:
Do we have a way to initiate shutdown from there?